### PR TITLE
chore: update GitHub branch references to use HEAD

### PR DESCRIPTION
**Pull request by thinkingserious**
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by childish-sambino** [comment]
I think most of the links can be swapped for relative paths which don't require the branch/tag name.

**Comment by thinkingserious** [comment]
Good call, updating ...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,27 +77,27 @@ Next, update your Environment with your [SENDGRID_APIKEY](https://app.sendgrid.c
 ##### Execute: #####
 
 - Check out the documentation for [Web API v3 endpoints](https://sendgrid.com/docs/API_Reference/Web_API_v3/index.html).
-- Review the corresponding [examples](https://github.com/sendgrid/sendgrid-csharp/blob/master/examples).
-- You can add your test code to our [Example Project](https://github.com/sendgrid/sendgrid-csharp/blob/master/ExampleCoreProject/Example.cs).
+- Review the corresponding [examples](examples).
+- You can add your test code to our [Example Project](ExampleCoreProject/Example.cs).
```
Comment (childish-sambino) [inline CONTRIBUTING.md:81]: Old link doesn't work.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-![SendGrid Logo](https://github.com/sendgrid/sendgrid-python/raw/master/twilio_sendgrid_logo.png)
+![SendGrid Logo](https://github.com/sendgrid/sendgrid-python/raw/HEAD/twilio_sendgrid_logo.png)
```
Comment (childish-sambino) [inline README.md:1]: Can this be relative, too?

File: src/SendGrid/Helpers/Mail/README.md [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
-Run the [Example Project](https://github.com/sendgrid/sendgrid-csharp/tree/master/ExampleCoreProject) (make sure you have set your Environment variable to include your SENDGRID_API_KEY).
+Run the [Example Project](ExampleCoreProject) (make sure you have set your Environment variable to include your SENDGRID_API_KEY).
 
 Click on the Example project, then click the `Start` button in the menu.
 
 ## Usage
 
-- See the [example](https://github.com/sendgrid/sendgrid-csharp/tree/master/ExampleCoreProject/Example.cs) for a complete working example.
+- See the [example](ExampleCoreProject/Example.cs) for a complete working example.
```
Comment (childish-sambino) [inline src/SendGrid/Helpers/Mail/README.md:11]: Old links don't work.

File: USE_CASES.md [hunk 4]
```diff
@@ -858,14 +858,14 @@ var client = new SendGridClient(options);
 <a name="domain-authentication"></a>
 # How to Setup a Domain Authentication
 
-You can find documentation for how to setup a domain authentication via the UI [here](https://sendgrid.com/docs/ui/account-and-settings/how-to-set-up-domain-authentication/) and via API [here](https://github.com/sendgrid/sendgrid-nodejs/blob/master/packages/client/USAGE.md#sender-authentication).
+You can find documentation for how to setup a domain authentication via the UI [here](https://sendgrid.com/docs/ui/account-and-settings/how-to-set-up-domain-authentication/) and via API [here](https://github.com/sendgrid/sendgrid-nodejs/blob/HEAD/packages/client/USAGE.md#sender-authentication).
```
Comment (childish-sambino) [inline USE_CASES.md:861]: nodejs link?

File: README.md (hunk 2 above)
Comment (thinkingserious) [reply README.md:1]: Yes, we just need to add the logo file to this repo.

File: CONTRIBUTING.md (hunk 1 above)
Comment (thinkingserious) [reply CONTRIBUTING.md:81]: Fixing ...

File: src/SendGrid/Helpers/Mail/README.md (hunk 3 above)
Comment (thinkingserious) [reply src/SendGrid/Helpers/Mail/README.md:11]: Fixing ..

File: USE_CASES.md (hunk 4 above)
Comment (thinkingserious) [reply USE_CASES.md]: Fixing ...

File: src/SendGrid/Helpers/Mail/README.md [hunk 5]
```diff
@@ -2,11 +2,11 @@
 
 # Quick Start
 
-Run the [Example Project](https://github.com/sendgrid/sendgrid-csharp/tree/master/ExampleCoreProject) (make sure you have set your Environment variable to include your SENDGRID_API_KEY).
+Run the [Example Project](ExampleCoreProject) (make sure you have set your Environment variable to include your SENDGRID_API_KEY).
```
Comment (childish-sambino) [inline src/SendGrid/Helpers/Mail/README.md:5]: Links must be relative to the current dir.

**Pull request merged**